src/structures/APIMessage.js

```diff
@@ -91,8 +91,16 @@ class APIMessage {
     const disableMentions = typeof this.options.disableMentions === 'undefined' ?
       this.target.client.options.disableMentions :
       this.options.disableMentions;
-    if (disableMentions) {
+    if (disableMentions === 'all') {
       content = Util.removeMentions(content || '');
+    } else if (disableMentions === 'everyone') {
+      content = (content || '').replace(/@([^<>@ ]*)/gsmu, (match, target) => {
+        if (target.match(/^[&!]?\d+$/)) {
+          return `@${target}`;
+        } else {
+          return `@\u200b${target}`;
+        }
+      });
```

Comment:
this would also replace `@somerandomtext` with `@<zws>somerandomtext` , (not sure how to go about fixing that, or if it would *need* fixing), thought it should be noted

Reply:
I don't believe this should be necessary to fix since it just assumes if the target isn't a valid mention of a person or a role mention, it has the potential to be unsanitized content